api/v1beta1/hostedcluster_conditions.go

```diff
@@ -155,6 +159,8 @@ const (
 	NotFoundReason            = "NotFound"
 	WaitingForAvailableReason = "waitingForAvailable"
 	SecretNotFoundReason      = "SecretNotFound"
+	Done                      = "Done"
+	WaitingForGracePeriod     = "waitingForGracePeriod"
```

Comment:
nit:
```suggestion
	DoneReason                      = "Done"
	WaitingForGracePeriodReason     = "waitingForGracePeriod"
```

we don't normally use a specific reason when the condition status is `True`, so you could could just use `AsExpectedReason` instead of Done